espflash/src/flasher/mod.rs

```diff
@@ -1382,3 +1409,14 @@ fn detect_chip(connection: &mut Connection, use_stub: bool) -> Result<Chip, Erro
         }
     }
 }
+
+fn detect_sdm(connection: &mut Connection) -> Result<bool, Error> {
```

Comment:
Can we detect sd mode in the normal detection routine, instead of doing it twice?

Reply:
I.e, inside `connection.begin()`, can we return the download mode type there?